tests/fsharp/core/nameof/preview/test.fsx

```diff
@@ -103,6 +103,14 @@ type BasicNameOfTests() =
         let b = nameof(this.MemberMethod)
         Assert.AreEqual("MemberMethod", b)
 
+    member this.``can get name of instance member via unchecked`` () =
+        let b = nameof(Unchecked.defaultof<BasicNameOfTests>.MemberMethod)
+        Assert.AreEqual("MemberMethod", b)
+
+    member this.``can get name of method type parameter``<'TTT> () =
```

Comment:
Should also add this, perhaps as a test in another file

```fsharp
type C<'DootDoot>() =
    member _.Name() = nameof<'DootDoot>
```

Reply:
Yes agreed